Transactions.md

```diff
@@ -1,182 +1,292 @@
+  * `txin[0]` sequence: 0xFFFFFFFE
+  * `txin[0]` script bytes: 0
+  * `txin[0]` witness: `0 <signature_for_offer_pubkey> <signature_for_accept_pubkey>`
+
+The output script public keys and `contract_maturity` are negotiated in the offer and accept messages.
+
+There will be one CET for every possible outcome, the output values correspond to such an outcome and are negotiated in the offer message.
+
+## CET Outputs
+
+The Offerer/Initiator's output comes first, the Accepter's output comes second.
+
+Note that this will likely change in the future.
+
+If either party receives less than the dust limit of `1000 satoshis` for this outcome, then their output is not produced.
+
+### Offerer Output
+
+This output sends funds won by the offerer corresponding to this CET's outcome to the offerer's final address specified in the offer message.
+
+### Accepter Output
+
+This output sends funds won by the accepter corresponding to this CET's outcome to the accepter's final address specified in the accept message.
+
+# Refund Transaction
+
+The refund transaction is exactly the same as a [Contract Execution Transaction](#contract-execution-transaction) except that its locktime is `contract_timeout` (as negotiated in the offer message) instead of `contract_maturity` and the output values for the offerer and the accepter are their total collateral values in the offer and accept message respectively.
+
+# Fees
+
+### Fee Calculation
+
+All fee calculations for all transactions is based on the fee rate specified in the offer message and the *expected weight* of the transaction in question.
+
+The actual and expected weights vary for several reasons:
+
+* Bitcoin uses DER-encoded signatures, which vary in size.
+* Bitcoin also uses variable-length integers, so a large number of outputs will take 3 bytes to encode rather than 1.
+* The offerer output may be below the dust limit.
+* The accepter output may be below the dust limit.
+
+Thus, a simplified formula for *expected weight* is used, which assumes:
+
+* Signatures are 72 bytes long.
+* There are a small number of outputs (thus 1 byte to count them).
+* There are always both an offerer output and an accepter output.
+
+This yields the following *expected weights* (details of the computation below):
+
+```
+Funding Transaction weight:    286 + 4 * total_change_length + 272 * num_inputs
+CET/Refund Transaction weight: 500 + 4 * total_output_length
+```
+
+### Fee Payment
+
+All fees are paid for in the [Funding Transaction](#funding-transaction) so that the funding output's value is inflated so that the outputs of CETs and the refund transaction are the exact amounts specified in the offer message's contract information (or total collateral specified in the offer and accept messages for the refund transaction) with no fees subtracted.
+
+All fees are currently paid evenly between the two parties, though this will change in a future version.
```

Comment:
One party can make the cost of aggregating its utxos spread unevenly by the other one. Alice declares 10 inputs, Bob only 2, assuming spends are equal, Bob will pays for 4 of Alice spends ?

A partial counter-measure, left to implementations, is to reject DLC offer if number of inputs are blatantly unbalanced.

Reply:
So as I've kind of alluded to here, I think the plan for v0.1 is to not have users pay fees evenly but rather to have each user pay for their own inputs and change output (and the rest evenly), but I didn't want to clutter this PR with that discussion so I'm hoping to open a PR once this gets merged with that part of the specification, I'll open an issue for this so I don't forget.

Reply:
Here it is including some explanation of what change will be needed :) https://github.com/discreetlogcontracts/dlcspecs/issues/58

Reply:
All those fee calculations are really a rabbit hole of petty details that can go wrong.
In the [payjoin proposal](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki), I had a similar issue, and the fee calculation was going to take like 60% of the complexity of the protocol.

I ended up managing to remove the problem entirely by letting the flexibility of choosing what fee to pay for all parties. All parties decide what they want to pay.

In the case nobody wants to pay any fee, and that would make the transaction unbroadcastable, then there is just an error.

In most case, there is no need to maintain the same feerate across funding, CET and refund. Why forcing it? This make the protocol more complicated.
On top of it, it can actually be dangerous! You want the CET to have a higher fee rate than the refund. If the mempool is full and they both have the same feerate and the CET can't propagate on time, the refund would cancel it.